test/lib/matrix4.spec.js

```diff
@@ -289,58 +291,34 @@ test('Matrix4.translate', t => {
 //   t.end();
 // });
 
-// test('Matrix4.rotateAxis', t => {
-//   t.equals(typeof Matrix4.prototype.rotateAxis, 'function');
-//   const v = [1, 2, 3];
-//   const len = Math.sqrt(1 * 1 + 2 * 2 + 3 * 3);
-//   const theta = Math.PI / 4;
-//   const m = new Matrix4();
-
-//   v[0] /= len;
-//   v[1] /= len;
-//   v[2] /= len;
+test('Matrix4.rotateAxis', t => {
+  t.equals(typeof Matrix4.prototype.rotateAxis, 'function');
+  const v = new Vector3(1, 2, 3).normalize();
+  const theta = Math.PI / 4;
+  const m = new Matrix4();
 
-//   const ans = m.rotateAxis(theta, v);
-//   t.ok(abs(ans.n11 - 0.7280277013778687) < delta);
-//   t.ok(abs(ans.n12 - -0.525104820728302) < delta);
-//   t.ok(abs(ans.n13 - 0.4407272934913635) < delta);
-//   t.ok(abs(ans.n14 - 0) < delta);
-//   t.ok(abs(ans.n21 - 0.6087885979157627) < delta);
-//   t.ok(abs(ans.n22 - 0.7907905578613281) < delta);
-//   t.ok(abs(ans.n23 - -0.06345657259225845) < delta);
-//   t.ok(abs(ans.n24 - 0) < delta);
-//   t.ok(abs(ans.n31 - -0.3152016404063445) < delta);
-//   t.ok(abs(ans.n32 - 0.3145079017103789) < delta);
-//   t.ok(abs(ans.n33 - 0.8953952789306641) < delta);
-//   t.ok(abs(ans.n34 - 0) < delta);
-//   t.ok(abs(ans.n41 - 0) < delta);
-//   t.ok(abs(ans.n42 - 0) < delta);
-//   t.ok(abs(ans.n43 - 0) < delta);
-//   t.ok(abs(ans.n44 - 1) < delta);
-//   t.end();
-// });
+  const result = m.rotateAxis(theta, v);
+  const reference = new Matrix4([
+    0.7280277013778687, -0.525104820728302, 0.4407272934913635, 0,
```

Comment:
Comment on where the expected result comes from?

Reply:
> Comment on where the expected result comes from?

Yes agree that we can do a more inspired test case than recording the generated matrix. This was just quickly repurposing some old luma.gl code that was left over.